File: 1013-fibonacci-number/1013-fibonacci-number.java
class Solution {
    public int rec(int n,int dp[]){
        if(n==0||n==1){
            return n;
        }
        if(dp[n]!=-1){
            return dp[n];
        }
        dp[n]=rec(n-1,dp)+rec(n-2,dp);
        return dp[n];
    }
    public int fib(int n) {
        int dp[]=new int[n+1];
        Arrays.fill(dp,-1);
        return rec(n,dp);
    }
}

File: 1013-fibonacci-number/1013-fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if(n==0||n==1){
            return n;
        }
        int pr1=1;
        int pr2=0;
        for(int i=2;i<=n;i++){
            int c=pr1+pr2;
            pr2=pr1;
            pr1=c;
        }
        return pr1;
    }
};